packages/features/ee/billing/teams/index.ts

```diff
@@ -0,0 +1,61 @@
+import { Prisma } from "@prisma/client";
+
+import { IS_TEAM_BILLING_ENABLED } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+
+import { InternalTeamBilling } from "./internal-team-billing";
+import { StubTeamBilling } from "./stub-team-billing";
+import type { TeamBilling as _TeamBilling, TeamBillingInput } from "./team-billing";
+
+const stubTeam = { id: -1, metadata: {}, isOrganization: true, parentId: -1 };
+const teamBillingSelect = Prisma.validator<Prisma.TeamSelect>()({
+  id: true,
+  metadata: true,
+  isOrganization: true,
+  parentId: true,
+});
+
+export class TeamBilling {
```

Comment:
Yeah. I'm not sold on create either. My other option was "init" for initialize or setup. 

```
const teamBilling = TeamBilling.init(team)
const teamBilling = TeamBilling.findAndInit()
```

Reply:
I agree with @joeauyeung here. This feels more like a repository rather than a team domain object. We should rename imho.

In DDD, for example, a repository would be used to return data for a team. Then, using just a constructor, static methods like we have here or a factory, you take that data and construct a team object with it. Mixing the two responsibilities here is misleading to the caller.

Reply:
@keithwillcode would you suggest we rename this to TeamBillingRepository and implement a proper constructor? Getting rid of the static methods?

Reply:
Yeah since it has a bunch of prisma calls in it.

Reply:
Addressed [in here](https://github.com/calcom/cal.com/pull/15366/commits/32ced4003b9493af7faa8d2827385a24dcf425c8) @keithwillcode

Reply:
Instead of renaming I've extracted the repository code into its own repo file and consume it in the static methods to avoid rewriting all the test code and implementations. cc @sean-brydon